pkg/controllers/organization/dex_controller.go

```diff
@@ -67,25 +68,33 @@ func (r *DexReconciler) SetupWithManager(name string, mgr ctrl.Manager) error {
 }
 
 func (r *DexReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	var org = new(greenhousesapv1alpha1.Organization)
-	if err := r.Get(ctx, req.NamespacedName, org); err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+	return lifecycle.Reconcile(ctx, r.Client, req.NamespacedName, &greenhousesapv1alpha1.Organization{}, r, nil)
+}
+
+func (r *DexReconciler) EnsureDeleted(_ context.Context, _ lifecycle.RuntimeObject) (ctrl.Result, lifecycle.ReconcileResult, error) {
+	return ctrl.Result{}, lifecycle.Success, nil // nothing to do in that case
+}
+
+func (r *DexReconciler) EnsureCreated(ctx context.Context, object lifecycle.RuntimeObject) (ctrl.Result, lifecycle.ReconcileResult, error) {
+	org, ok := object.(*greenhousesapv1alpha1.Organization)
+	if !ok {
+		return ctrl.Result{}, lifecycle.Failed, errors.Errorf("RuntimeObject has incompatible type.")
 	}
 
 	// Ignore organizations without OIDC configuration.
 	if org.Spec.Authentication == nil || org.Spec.Authentication.OIDCConfig == nil {
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, lifecycle.Pending, nil
```

Comment:
Is `Pending` being returned so that the `Ready` status is not set?

Reply:
Done, I switched to lifecycle.Success in that case.

Reply:
I move it to the predicate to filter reconciliation.